qiime/sdk/visualizer.py

```diff
@@ -158,6 +158,12 @@ def _bind_executors(self):
         # TODO `async` execution has some problems with garbage-collection in
         # the subprocess given certain inputs. Needs further investigation.
         def async_wrapper(*args, **kwargs):
+            # TODO handle this better in the future, but stop the massive error
+            # caused by MacOSX async runs for now.
+            import matplotlib as plt
```

Comment:
Can you wrap this in a try/except for `ImportError` so that matplotlib is an optional dependency? Then matplotlib can be removed from .travis.yml and setup.py.